compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/StreamingCodeDesugar.java

```diff
@@ -245,8 +256,21 @@ public void visit(BLangStreamingQueryStatement queryStmt) {
             isInJoin = true;
             BLangStreamingInput streamingInput = (BLangStreamingInput) joinStreamingInput.getStreamingInput();
             createStreamAliasMap(streamingInput);
-            rhsStream = (BLangSimpleVarRef) joinStreamingInput.getStreamingInput().getStreamReference();
-            lhsStream = (BLangSimpleVarRef) queryStmt.getStreamingInput().getStreamReference();
+            // If the node kind is INVOCATION, considering that as a table join.
+            if (queryStmt.getStreamingInput().getStreamReference().getKind() == NodeKind.INVOCATION) {
```

Comment:
What if there is a variable global variable (which is the table) defined and it is used when joining with streams? I don't think considering it only as Invocation is wrong.

Reply:
Fixed in b7e0182